aries_cloudagent/wallet/did_method.py

```diff
@@ -20,11 +20,11 @@ class DIDMethod(Enum):
     """DID Method class specifying DID methods with supported key types."""
 
     SOV = DIDMethodSpec(
```

Comment:
Just as a matter of interest -- should there be other types of DID methods here now or in the future?  For example, should `did:indy` be here since we want to use it Real Soon Now.  If not, has consideration been given to how it would be added?  Also wondering about `did:peer`.  What if we start supporting another VDR (e.g. cheqd.io)?

I'm clueless at the this level, but I'm surprised that there are hardcoded references in ACA-Py code to specific DID Methods and cryptographic suites (e.g. ED25519 throughout the code) vs. them being variables with late binding during deployments.  I'm guessing I'm confused about this?

Reply:
Indeed, this is exactly what we're seeking to be able to do. #1955 specifically addresses turning the set of known DID Methods from a static enum (as seen on these lines) into something that can be determined on start up with plugins getting a chance to add more methods. This PR does a similar "pluggification" or late binding as you put it for key types as the other PR does for DID methods.

While we are making methods and key types pluggable values, to limit breaking changes, the previously statically defined methods and key types are retained as statically declared values inserted into a dynamic registry by default.

At some future date, I think it makes sense for ACA-Py itself to only directly "know" about DID Peer and then all other methods are plugged in or configurable as built in plugins. This PR and #1955 are intended as incremental steps in that direction. Additional feedback on the approach is welcome :slightly_smiling_face: